Meta/CMake/lagom_compile_options.cmake

```diff
@@ -6,6 +6,7 @@ add_compile_options(-fsigned-char)
 add_compile_options(-g1)
 add_compile_options(-ggnu-pubnames)
 add_compile_options(-O2)
+add_compile_options(-march=native -mtune=native)
```

Comment:
This is going to need a closer look from @ADKaster, as building with `-march=native` effectively makes it impossible to distribute a piece of software.

I'm also worried about `-march=native`'s interaction with `ccache` on GitHub actions. When we tried enabling `-march=native` for Toolchain builds, jobs started failing randomly because not all runners supported the same CPU extensions:  #7346.

Reply:
> building with `-march=native` effectively makes it impossible to distribute a piece of software.

True, though I don't think distributing binaries is a primary concern.

> I'm also worried about `-march=native`'s interaction with `ccache` on GitHub actions. When we tried enabling `-march=native` for Toolchain builds, jobs started failing randomly because not all runners supported the same CPU extensions:  #7346.

This is a concern I had not considered. How about a NO_NATIVE_ARCH flag for Cmake that downstream packagers and CI can use?

> `-march=native` meaning different things in cached and newly built object files could also lead to ABI incompatibilities.

I'm pretty sure that only applies to vector passing across ABI boundaries, which is unspecified by Itanium. So I'm certain the compilers won't do it unless they control both caller and callee, i.e. not for exported symbols in shared libraries.

Reply:
if we use -march=native, then the compiler may generate AVX-512 or other advanced SIMD instructions, and store them in the ccache. When those object files are linked and executed on a machine without those instructions, we'll get an invalid instruction crash. This is an issue for both the Lagom and the Serenity builds, because we execute the code generators using the same Lagom options. So the ABI concern here is more than just function call boundaries, it's the actual machine code that is generated and executed. The CI machines are a massive mixed bag of old and new hardware that we can't predict without manually setting up our own build farm.

Rather than adding a NO_NATIVE_ARCH flag, I would rather break out "only one build mode" rule, and add "Omega Debug", "Debug (standard)" and "Release" build modes, with march native/mtune native being set only in the Release mode

Reply:
Sounds good as well, but why the distinction between Omega debug and Debug (standard)? Also, the whole point of this PR is to make -march=native the standard for the common case (i.e. developers and personal projects which use Lagom) so we'd have to make sure that's the case.

Reply:
standard debug is `-g1 -O2`, while omega debug is `-ggdb3 -Og` or `-O0`, which you almost always have to reach for when using a debugger.

Reply:
As for using this in the standard case, what is the impact on compile times? When I was testing this locally the other day it seemed to me that using -march=native -mtune=native took significantly longer to compile.

Reply:
> building with -march=native effectively makes it impossible to distribute a piece of software.

Keep in mind we do distribute `js`.